code/modules/atmospherics/machinery/components/components_base.dm

```diff
@@ -90,27 +90,46 @@
 			P.build_pipeline(src)
 
 /obj/machinery/atmospherics/components/proc/nullifyPipenet(datum/pipeline/reference)
+	// Crash when we call a null pipenet here. Pretty straightforward.
 	if(!reference)
 		CRASH("nullifyPipenet(null) called by [type] on [COORD(src)]")
-	var/i = parents.Find(reference)
-	reference.other_airs -= airs[i]
+
+	// Disconnect each air mixtures from the other_airs list in the pipeline. 
+	// Also nullifies the specific entry in the parents list. This is mainly used when a machinery is rotated.
+	for (var/i in 1 to parents.len)
+		if (parents[i] == reference)
+			reference.other_airs -= airs[i]
+			parents[i] = null
+
+	// Deletes any reference of the atmos machinery as listed in the other_atmosmch list from the pipeine.
 	reference.other_atmosmch -= src
+
+	// Cleans pipenet up
+	if(!length(reference.other_atmosmch) && !length(reference.members))
+		if(QDESTROYING(reference))
+			CRASH("nullifyPipenet() called on qdeleting [reference]")
+		qdel(reference)
+
 	/**
 	 *  We explicitly qdel pipeline when this particular pipeline
 	 *  is projected to have no member and cause GC problems.
 	 *  We have to do this because components don't qdel pipelines
 	 *  while pipes must and will happily wreck and rebuild everything
 	 *	again every time they are qdeleted.
 	 */
-	if(!length(reference.other_atmosmch) && !length(reference.members))
-		if(QDESTROYING(reference))
-			parents[i] = null
-			CRASH("nullifyPipenet() called on qdeleting [reference] indexed on parents\[[i]\]")
-		qdel(reference)
-	parents[i] = null
 
-/obj/machinery/atmospherics/components/returnPipenetAir(datum/pipeline/reference)
-	return airs[parents.Find(reference)]
+/obj/machinery/atmospherics/components/returnPipenetAirs(datum/pipeline/reference)
+	var/list/returned_air = list()
+	// We return a list filled with a single null entry if we dont have a proper parents list.
+	// This will get picked up by addMachineryMember to call a stack_trace.
+	if (!parents || !length(parents))
+		returned_air += null
+		return returned_air
```

Comment:
For future reference, length() works just fine on nulls, so you don't need a safety check here. That said, this behavior should be handled by the check in addMachineryMember(), so you can safely toss this.

Reply:
Tossed.